babel.config.js

```diff
@@ -1,50 +1,9 @@
 const config = {
-	presets: [
-		[
-			'@babel/env',
-			{
-				useBuiltIns: 'entry',
-				corejs: 2,
-				// Exclude transforms that make all code slower, see https://github.com/facebook/create-react-app/pull/5278
-				exclude: [ 'transform-typeof-symbol' ],
-			},
-		],
-		'@babel/react',
-	],
-	plugins: [
-		'@babel/plugin-proposal-class-properties',
-		'@babel/plugin-proposal-export-default-from',
-		'@babel/plugin-proposal-export-namespace-from',
-		'@babel/plugin-syntax-dynamic-import',
-		[
-			'@babel/transform-runtime',
-			{
-				corejs: false, // we polyfill so we don't need core-js
-				helpers: true,
-				regenerator: false,
-				useESModules: false,
-			},
-		],
-	],
+	extends: require.resolve( '@automattic/calypso-build/babel.config.js' ),
```

Comment:
`extends` or `preset`? See https://github.com/Automattic/wp-calypso/pull/32024/files#r272567294

Reply:
Seems like presets are configs and/or sets of plugins published from somewhere else and shared across projects e.g. within the company, based on reading:

https://babeljs.io/docs/en/presets
https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/user-handbook.md#making-your-own-preset

... I don't find much reading about nature of extends, are they used mainly to extend config within the same project?
 
https://babeljs.io/docs/en/options#extends

Seems like to be a preset, our package would need to be `@automattic/babel-preset-*` — or is that only for name normalization to work and any name would work otherwise? Are presets expected to export only Babel config, where as our package carries Babel config, Webpack config and more and hence we should use `extends`?

I also had a look at https://babeljs.io/docs/en/options#merging but didn't really get any smarter. 😅

Reply:
Yeah, sorry, I should've mentioned the way that both work (glad that you [found out yourself](https://github.com/Automattic/jetpack/pull/11949/files#r273077528)). The question wasn't so much which of them worked correctly -- I'm pretty sure both do in this case -- but rather about which one we'd like to use semantically. 

`preset` is more elegant in this case, but potentially sets a misleading precedent if others want to override parts of the config we're getting from `@automattic/calypso-build/babel.config.js`, much like we do in Calypso's top-level `babel.config.js`.

Thus inclined to stick with `extends` for now.

Reply:
`extends` sounds good yep. 👍 

Ping @gziolo in case you have some wisdom about this. :-)

Reply:
We don't use this feature in Gutenberg 🤷‍♂️